src/modules/mtg-search.ts

```diff
@@ -111,20 +120,134 @@ export class MTGSearch {
+            this.sendSingleCardResult(msg, cardMatchFromName);
+          } else {
+            // If there are multiple cards, we provide up to the top 5 results and ask the user to refine.
+            this.sendMultipleCardResults(msg, cardResults);
           }
-          this.HBot.sendMessage(msg.chat.id, message, { parse_mode: 'HTML', disable_web_page_preview: true });
         }
       });
   }
 
+  /**
+   * Function to search the ScryFall API using an existing cardId. If
+   * a card is found, we will send it to the user.
+   * @param msg - The initial message sent to the bot. It allows us to send a message back.
+   * @param cardId - The Id of the card we want to send to the user.
+   */
+  private getMtgCardById(msg: TelegramBot.Message, cardId: string): void {
+    Cards.byId(cardId).then(
+      (result: Card) => {
+        this.sendSingleCardResult(msg, result);
+      },
+      (error) => {
+        this.HBot.sendMessage(msg.chat.id, `${error.response.body.details}`);
+      },
+    );
+  }
+
+  /**
+   * Function to send user a single MTG card. Data is sent as a
+   * picture with the caption of the USD price, if applicable.
+   * @param msg - The initial message sent to the bot. It allows us to send a message back.
+   * @param card - The Card data fetched from ScryFall.
+   */
+  private sendSingleCardResult(msg: TelegramBot.Message, card: Card): void {
+    const priceCaption: string = card.usd ? `USD Price: $${card.usd}` : null;
+    if (card.card_faces) {
+      // Multi faced card case
+      card.card_faces.forEach((cardFace: any, index: number) => {
+        this.HBot.sendPhoto(msg.chat.id, `${cardFace.image_uris.normal}`,
+        {
+          caption: index === card.card_faces.length - 1 ? priceCaption : undefined,
+          reply_markup: {
+            remove_keyboard: true,
+            selective: true,
+          },
+          reply_to_message_id: msg.message_id,
+        });
+      });
+    } else {
+      this.HBot.sendPhoto(msg.chat.id, `${card.image_uris.normal}`,
+        {
+          caption: priceCaption,
+          reply_markup: {
+            remove_keyboard: true,
+            selective: true,
+          },
+          reply_to_message_id: msg.message_id,
+        });
+    }
+  }
+
+  /**
+   * Function to let the user know their search result found more than one
+   * result when using their query. We can display up to 120 results to the
+   * user for them to select from along with a message asking for a refined
+   * query.
+   * @param msg - The initial message sent to the bot. It allows us to send a message back.
+   * @param userCards - List of MTG cards that were returned from the user's search query.
+   */
+  private sendMultipleCardResults(msg: TelegramBot.Message, userCards: Card[]): void {
+    let message: string = `Multiple cards found (${userCards.length}). Did you mean one of the following? If not, try refining or changing your query.\n`;
+    if (userCards.length > 120) {
```

Comment:
120 seems like a lot...maybe limit to 5 for sanity's sake?  Dunno what the interface for this looks like though...

Reply:
The goal here is to provide the user with the maximum amount of choices to prevent having to search again. If the user searched for "fire", we have all the results from the API already (meaning this does not affect performance) and since the keyboard sent to the user isn't visible by anyone but themselves, they can scroll through the entire keyboard or simply dismiss it. Without this, there is no way to present these options in a better format given limitations from the Bot API.

Reply:
I figured it might be a scrolling thing.  Sounds good.  I worried it might be a big ol block of shit on the screen